ledger/participant-integration-api/src/main/scala/platform/apiserver/services/ApiSubmissionService.scala

```diff
@@ -155,9 +160,22 @@ private[apiserver] final class ApiSubmissionService private[services] (
             }
         case _: CommandDeduplicationDuplicate =>
           metrics.daml.commands.deduplicatedCommands.mark()
-          val exception = duplicateCommandException
-          logger.debug(exception.getMessage)
-          Future.failed(exception)
+          errorCodesVersionSwitcher.chooseAsFailedFuture(
+            v1 = {
+              val exception = duplicateCommandException
+              logger.debug(exception.getMessage)
+              exception
+            },
+            v2 = {
+              val exception = RejectionGenerators.duplicateCommand(
+                logger = logger,
+                loggingContext = implicitly[LoggingContext],
+                correlationId = CorrelationId.none,
+              )
+              logger.debug(exception.getMessage)
```

Comment:
The error instantiation already logs.

Reply:
Ok, removing.

Added it originally because this error instantiation seems to be logging to INFO. Also the messages might be different. Haven't checked the actual values though